include/boost/geometry/algorithms/detail/overlay/handle_colocations.hpp

```diff
@@ -507,6 +507,46 @@ inline void discard_interior_exterior_turns(Turns& turns, Clusters& clusters)
     }
 }
 
+template<typename Turns, typename Clusters>
+inline void discard_start_turns(Turns& turns, Clusters& clusters)
+{
+    for (auto& nv : clusters)
+    {
+        cluster_info& cinfo = nv.second;
+        auto& indices = cinfo.turn_indices;
+        std::size_t start_count{0};
```

Comment:
Out of curiosity, why not `std::size_t start_count = 0;` ?

Reply:
Habit, it the office we're preferring the new uniform initialization.

What do you think?

Reply:
I used to be very keen in preferring the uniform initialization heavily and everywhere, until I read the abseil guidelines and tips, in particular https://abseil.io/tips/88

> First, “uniform” is a stretch: there are cases where ambiguity still exists (for the casual reader, not the compiler) in what is being called and how.
> 
> ```
> std::vector<std::string> strings{2}; // A vector of two empty strings.
> std::vector<int> ints{2};            // A vector containing only the integer 2.
> ```

It's a pity the uniform initialization is not corner-case free improvement to C++.

Reply:
In this particular case I'd use `= 0` (also for numeric classes) because in such case the variable represents `0`.

This is not a case when an object of a class is created by passing some abstract which may describe something like number of elements.

From the same reason I'd probably initialize the vector like the one below. In other words to express that the initializer list on the right represents the vector or that it is the vector (just like with `0` being the number):
```
std::vector<int> ints = {2};
```
This works right? ;)

And then for all other things I don't have an opinion. I'd probably use the old parentheses. But maybe that's just because it's what I'm familiar with. Maybe with the exception of avoiding double parentheses to deal with the most vexing parse.

So I guess I could also express this as a conservative stance where old initialization is used when possible as it was in the past and new brackets only for new features.

Reply:
Interesting visions, thanks.
I agree with the statement that it's not that intuitive, `int i{0}` is quite different from all other languages where it is something like `int i=0;`
I'll think about it. For now I will merge this, assuming these were not objections.